pom.xml

```diff
@@ -385,6 +445,48 @@ THE SOFTWARE.
         </plugins>
       </build>
     </profile>
+    <!--
+      Our Jenkinsfile has a Java 17 branch that tests a recent weekly release for two reasons:
+
+        1. We want to test on all versions of Java, but only recent weeklies support Java 17.
+
+        2. Testing against a recent weekly release ensures that the test harness can be compiled
+           against the latest core APIs.
+
+      When we are testing against a recent weekly that requires Java 11, the bytecode version
+      check naturally fails. To work around this, we relax the bytecode version check
+      when we are executing the Java 17 branch in the CI environment.
+
+      TODO When the minimum Jenkins baseline is bumped past 2.357, this can be deleted.
+    -->
+    <profile>
+      <id>test-recent-cores-on-ci</id>
+      <activation>
+        <jdk>17</jdk>
+        <property>
+          <name>env.CI</name>
+        </property>
```

Comment:
But then someone who happens to be running locally with Java 17 but has not customized the Jenkins version would get the looser Enforcer semantics of 11 and may accidentally depend on a Java 11 API. The CI build would catch this before it is merged and released, but I think local and CI builds should be as close as possible. I like restricting this to CI builds since that is the only place where we are programmatically using a recent weekly version.

Reply:
> local and CI builds should be as close as possible

Well of course that was my point as well, from another perspective. :grin: Anyway, this is fine.

Reply:
Right, someone who is running locally with `jenkins.version` would need to set `CI=true` to get this logic. I dunno if there is a good solution here. Really what I need is a way to activate the profile if the version number is greater than or equal to some value, but I can't think of a way to do that with Maven profile activation.

Reply:
> activate the profile if the version number is greater than or equal to some value

Yeah that is moving into Gradle territory I think.